### PR TITLE
CHK-522: Fix installments not being properly sent to gateway

### PR DESCRIPTION
**Pull request by jeffersontuc**
#### What problem is this solving?

The aim of this PR is to fix the installments value not sent properly. That was happening due to the payment object missing the right key name, as @lucasecdb said on the [JIRA card](https://vtex-dev.atlassian.net/browse/CHK-522?atlOrigin=eyJpIjoiZjAyYzQyMDFjMmM4NGYzM2I4NTA1NDkxNzQwOTM2YTUiLCJwIjoiaiJ9).

#### How should this be manually tested?

- [Add an item to cart](https://installment--checkoutio.myvtex.com/cart/add?sku=289)
- Proceed to the checkout (you can use a second purchase email)
- Pick a credit card as a payment method
`4040240009008936`
`02/22`
`066`
- Choose an installment option
- Finish the purchase
- Copy the order id
- [Go to the admin](https://checkoutio.myvtex.com/admin) 

You can use the following url to get directly to the order page (don't forget to replace `orderId` by the one copied earlier) https://checkoutio.myvtex.com/admin/checkout/#/orders/{orderId}-01

- Click to see the transaction details (as in the screenshot)
- Check if you can see the right amount of installments on the transaction details (as in the screenshot)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

image

image

#### Type of changes

✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

I was thinking in how to test this, but I was not able to come up with an idea. If you know a way, just leave a comment.

### CODE REVIEW

**Review by lucasecdb — COMMENTED** [review]
do you think we need to update in the `merchantSellerPayments` array as well?

image

would be nice if we _didn't_ send the `installmentValue` too 😅

**Comment by lucasecdb** [comment]
oh, look like v6 _doesn't_ send this `merchantSellerPayments` array! so we could "get rid" of it altogether

**Comment by lucasecdb** [comment]
oh, we could also fix the interest rate, the correct property is `installmentsInterestRate` instead of `interestRate`, could you do that too @jeffersontuc?

**Comment by jeffersontuc** [comment]
> oh, we could also fix the interest rate, the correct property is `installmentsInterestRate` instead of `interestRate`, could you do that too @jeffersontuc?

Sure!

File: react/PlaceOrder.tsx [hunk 1]
```diff
@@ -107,11 +107,14 @@ const PlaceOrder: React.FC = () => {
                 return null
               }
 
+              const { merchantSellerPayments, ...payment } = transactionPayment
+
               return {
-                ...transactionPayment,
                 ...merchantPayment,
+                ...payment,
```
Comment (wdsrocha) [inline react/PlaceOrder.tsx]: Can you tell if there is any risk of overwriting?

Comment (jeffersontuc) [reply react/PlaceOrder.tsx]: They have basically the same values, but I'll move back the payment object just to be sure

**Review by wdsrocha — APPROVED** [review]
LGTM, but I think that besides running the e2e tests before deploying, it would be good to send this PR to the functional validation again (even if it's already in production by monday)

**Review by silviasfon — APPROVED** [review]

**Pull request merged**